### ISSUE TITLE
string t, n, y are parsed as boolean

### ISSUE THREAD

**Issue by kimim** [techascent/tech.ml.dataset#166]
I have a CSV table as bellow.

class | cap-color | bruises | odor
-- | -- | -- | --
p | n | t | p
e | y | t | a
e | w | t | l
p | w | t | p
e | g | f | n

I want to parse all the characters as string. But I got something like this:

    | class | cap-color | bruises | odor |
    |-------|-----------|---------|------|
    |     p |     false |    true |    p |
    |     e |      true |    true |    a |
    |     e |         w |    true |    l |
    |     p |         w |    true |    p |
    |     e |         g |   false |    n |

**Comment by cnuernber** [comment]
I explicitly did this as I remember from many years ago getting data from businesses where n,y,t,f should be parsed as booleans.  I wonder if that is too much of a special case.  For this file specifically it seems all columns should be strings and thus `{:parser-fn :string}` will work.

In the general case perhaps I overstepped here.  I am not certain.

**Comment by kimim** [comment]
Use :parser-fn :string is as expected. Maybe default to get string is more intuitive. How do you think about this suggestion? If that’s fine, may I try to create a PR for this issue.

**Comment by cnuernber** [comment]
I think your suggesting of removing the single-letter variants is wise and less confusing than the current behavior.  I would very much appreciate a PR.

**Comment by kimim** [comment]
after moving :boolean to the end, the parser process looks fine:
https://github.com/techascent/tech.ml.dataset/blob/8f1aa3d4423df2f9a41765de8a8961fe7773ea83/src/tech/v3/dataset/io/column_parsers.clj#L286

```
(def default-parser-datatype-sequence
  [:int16 :int32 :int64 :float64 :uuid
   :packed-duration :packed-local-date
   :zoned-date-time :string :text :boolean])
```

**Comment by kimim** [comment]
fixed by #167
